src/array_api_stubs/_draft/array_object.py

```diff
@@ -39,6 +43,7 @@ def device(self: array) -> Device:
         out: device
             a ``device`` object (see :ref:`device-support`).
         """
+        ...
 
     @property
     def mT(self: array) -> array:
```

Comment:
As a heads up: the [PEP 673](https://peps.python.org/pep-0673/) `Self` type (a special type variable that's always bound to the respective classes `self` parameter) is now also supported by mypy and the likes, so you could, if so desired, express this as:
```python
if TYPE_CHECKING:
    from typing_extensions import Self
    
 class Array(Protocol):
    def mT(self) -> Self:
```

Reply:
I have a note already
```
array = TypeVar("array", bound="Array")
# NOTE: when working with py3.11+ this can be ``typing.Self``.
```

But I'm happy to change it now. I was avoiding adding additional dependencies, even if they aren't runtime and 
are made by the core python devs.

Reply:
Then let's tackle this in a followup, since `Self` will need to be rendered in the docs as `Array`.